src/rust/src/packed_r_tree.rs

```diff
@@ -153,10 +181,8 @@ async fn read_http_node_items(
         .map_err(from_http_err)?;
 
     let mut node_items = Vec::with_capacity(length);
-    let buf = unsafe { std::slice::from_raw_parts_mut(node_items.as_mut_ptr() as *mut u8, len) };
-    buf.clone_from_slice(&bytes);
-    unsafe {
-        node_items.set_len(length);
+    for i in 0..length {
+        node_items.push(NodeItem::from_bytes(&bytes[i * size_of::<NodeItem>()..])?);
```

Comment:
Maybe better to be explicit about this range length rather than ".."  to make sure `from_bytes` isn't reading beyond the desired bounds.